src/Stratis.SmartContracts.CLR.Tests/SmartContracts/SystemContractsDictionary.cs

```diff
@@ -0,0 +1,257 @@
+    }
+
+    public uint GetQuorum(string group)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        return this.State.GetUInt32($"Quorum:{group}");
+    }
+
+    private void SetQuorum(string group, uint value)
+    {
+        this.State.SetUInt32($"Quorum:{group}", value);
+    }
+
+    private uint GetGroupNonce(string group)
+    {
+        return this.State.GetUInt32($"GroupNonce:{group}");
+    }
+
+    private void SetGroupNonce(string group, uint value)
+    {
+        this.State.SetUInt32($"GroupNonce:{group}", value);
+    }
+
+    public void AddSignatory(byte[] signatures, string group, Address address, uint newSize, uint newQuorum)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        Assert(newSize >= newQuorum, "The number of signatories can't be less than the quorum.");
+
+        Address[] signatories = this.GetSignatories(group);
+        foreach (Address signatory in signatories)
+            Assert(signatory != address, "The signatory already exists.");
+
+        Assert((signatories.Length + 1) == newSize, "The expected size is incorrect.");
+
+        uint nonce = this.GetGroupNonce(group);
+
+        this.VerifySignatures(signatures, $"{nameof(AddSignatory)}(Nonce:{nonce},Group:{group},Address:{address},NewSize:{newSize},NewQuorum:{newQuorum})");
+
+        System.Array.Resize(ref signatories, signatories.Length + 1);
+        signatories[signatories.Length - 1] = address;
+
+        this.SetSignatories(group, signatories);
+        this.SetQuorum(group, newQuorum);
+        this.SetGroupNonce(group, nonce + 1);
+    }
+
+    public void RemoveSignatory(byte[] signatures, string group, Address address, uint newSize, uint newQuorum)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        Assert(newSize >= newQuorum, "The number of signatories can't be less than the quorum.");
+
+        Address[] prevSignatories = this.GetSignatories(group);
+        Address[] signatories = new Address[prevSignatories.Length - 1];
+
+        int i = 0;
+        foreach (Address item in prevSignatories)
+        {
+            if (item == address)
+            {
+                continue;
+            }
+
+            Assert(signatories.Length != i, "The signatory does not exist.");
+
+            signatories[i++] = item;
+        }
+
+        Assert(newSize == signatories.Length, "The expected size is incorrect.");
+
+        uint nonce = this.GetGroupNonce(group);
+
+        this.VerifySignatures(signatures, $"{nameof(RemoveSignatory)}(Nonce:{nonce},Group:{group},Address:{address},NewSize:{newSize},NewQuorum:{newQuorum})");
+
+        this.SetSignatories(group, signatories);
+        this.SetQuorum(group, newQuorum);
+        this.SetGroupNonce(group, nonce + 1);
+    }
+
+    public bool IsWhiteListed(UInt256 codeHash)
+    {
+        Assert(codeHash != default(UInt256));
+
+        WhiteListEntry whiteListEntry = this.State.GetStruct<WhiteListEntry>(codeHash.ToString());
```

Comment:
```suggestion
        WhiteListEntry whiteListEntry = this.State.GetStruct<WhiteListEntry>($"CodeHash:{codeHash});
```
Always use namespace for keys and access state by getters/setters. No need to call ToString()

Reply:
Done.